src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -151,6 +152,58 @@ BuildAggregatePlan(Query *masterQuery, Plan *subPlan)
 }
 
 
+static Plan *
+BuildDistinctPlan(Query *masterQuery, Plan *topLevelPlan)
+{
+	bool distinctClausesHashable = true;
+	List *distinctClauseList = masterQuery->distinctClause;
+	Plan *plan = topLevelPlan;
+	List *aggregateTargetList = copyObject(masterQuery->targetList);
+	List *aggregateColumnList = pull_var_clause_default((Node *) aggregateTargetList);
+	ListCell *columnCell = NULL;
+
+	Assert(masterQuery->distinctClause);
+	Assert(!masterQuery->hasDistinctOn);
+
+	distinctClausesHashable = grouping_is_hashable(distinctClauseList);
+
+	foreach(columnCell, aggregateColumnList)
+	{
+		Var *column = (Var *) lfirst(columnCell);
+		column->varno = OUTER_VAR;
+	}
+
+	/*
+	 * Create group by plan with HashAggregate if all distinct
+	 * members are hashable, and query does not have distinct on
+	 * and query does not already have sort plan on distinct clauses.
+	 */
+	if (distinctClausesHashable)
+	{
+		/* finally create the plan */
+		const long rowEstimate = 10; /* this is taken from postgresql */
+		AttrNumber *distinctColumnIdArray = extract_grouping_cols(distinctClauseList,
+																  plan->targetlist);
+		Oid *distinctColumnOpArray = extract_grouping_ops(distinctClauseList);
+		uint32 distinctClauseCount = list_length(distinctClauseList);
+
+		plan = (Plan *) make_agg(aggregateTargetList, NIL, AGG_HASHED,
+								 AGGSPLIT_SIMPLE, distinctClauseCount,
+								 distinctColumnIdArray,
+								 distinctColumnOpArray, NIL, NIL,
+								 rowEstimate, plan);
+	}
+	else
+	{
+		plan = (Plan *) make_sort_from_sortclauses(masterQuery->distinctClause, plan);
+		plan = (Plan *) make_unique_from_distinctclause(plan,
```

Comment:
Can we use directly `make_unique_from_sortclauses` here instead of writing our own version `make_unique_from_distinctclause` ?

Reply:
Ohh, this `make_unique_from_distinctclause` is exactly same `make_unique_from_sortclauses`? I think we would like to keep original names with PostgreSQL, such as we do in `ruleutils_10.c`

Reply:
Ok, renamed it back to  ```make_unique_from_sortclauses```